app/src/main/java/net/sourceforge/opencamera/sensorremote/RemoteRpcServer.java

```diff
@@ -0,0 +1,183 @@
+package net.sourceforge.opencamera.sensorremote;
+
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.util.Log;
+
+import net.sourceforge.opencamera.MainActivity;
+import net.sourceforge.opencamera.MyDebug;
+
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RemoteRpcServer extends Thread {
+    private static final String TAG = "RemoteRpcServer";
+    private static final int SOCKET_WAIT_TIME_MS = 1000;
+    private static final String IMU_REQUEST_REGEX = "(imu\\?duration=)(\\d+)(&accel=)(\\d)(&gyro=)(\\d)";
+
+    private final ServerSocket mRpcSocket;
+    private final Properties mConfig;
+    private final RemoteRpcRequestHandler mRequestHandler;
+    private final AtomicBoolean mIsExecuting = new AtomicBoolean();
+    private final MainActivity mContext;
+
+    public RemoteRpcServer(MainActivity context) throws IOException {
+        mContext = context;
+        mConfig = RemoteRpcConfig.getProperties(context);
+        mRequestHandler = new RemoteRpcRequestHandler(context);
+        mRpcSocket = new ServerSocket(Integer.parseInt(mConfig.getProperty("RPC_PORT")));
+        mRpcSocket.setReuseAddress(true);
+        mRpcSocket.setSoTimeout(SOCKET_WAIT_TIME_MS);
+
+        if (MyDebug.LOG) {
+            Log.d(TAG, "Hostname " + getIPAddress());
+        }
+
+    }
+
+    public boolean isExecuting() {
+        return mIsExecuting.get();
+    }
+
+    public void stopExecuting() {
+        mIsExecuting.set(false);
+    }
+
+    private void handleRequest(String msg, PrintStream outputStream, BufferedOutputStream outputByte) {
+        // IMU remote control API
+        Pattern r = Pattern.compile(IMU_REQUEST_REGEX);
+        Matcher imuRequestMatcher = r.matcher(msg);
+        if (imuRequestMatcher.find()) {
+            long duration = Long.parseLong(imuRequestMatcher.group(2));
```

Comment:
Check and handle a malformed `msg`, at least it will help later to debug client applications. For example, if the protocol will be changed later in the newer versions of the app.